### PR TITLE
Bug 1199364 - Use revsion instead of revision_hash for resultsets

### PR DESCRIPTION
**Pull request by camd**
Bug 1199364 - Use revsion instead of revision_hash for resultsets

See individual commits for more info.  I sure apologize for the magnitude of this change.  At least MOST of it is in the tests and test data.  :)

[Review on Reviewable](https://reviewable.io/reviews/mozilla/treeherder/1322)

### CODE REVIEW

File: treeherder/client/thclient/client.py [hunk 1]
```diff
@@ -419,22 +426,15 @@ def init_data(self):
...
-    def add_revision_hash(self, revision_hash):
-        self.data['revision_hash'] = revision_hash
-
     def add_author(self, author):
         self.data['author'] = author
 
     def add_revisions(self, revisions):
-        if revisions:
-            self.data['revisions'] = revisions
+        for revision in revisions:
```
Comment (camd) [inline treeherder/client/thclient/client.py]: this has always been broken.  should have been adding the ``.data`` property...  :)

**Comment by edmorley** [comment]
Reviewed 30 of 40 files at r1, 15 of 15 files at r2.
Review status: all files reviewed at latest revision, 6 unresolved discussions.

---

*[tests/etl/test_job_loader.py, line 34 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0eX53sTmHsY24uyQU:-KC0eX53sTmHsY24uyQV:596319423) ([raw file](https://github.com/mozilla/treeherder/blob/363f8ae1d64b5d015496bbb07c6497534934b4a6/tests/etl/test_job_loader.py#L34)):*
Is the `result_set_stored` fixture parameter now unused here?

---

*[tests/etl/test_pushlog.py, line 40 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0cMlhIg7Zr1dKIb7y:-KC0cMlilAYdDY_lFfAw:-56371274) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/tests/etl/test_pushlog.py#L40)):*
`rev_to_push` is now unused, so this whole for loop can be removed I think?

---

*[tests/etl/test_pushlog.py, line 44 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0bvRolf6ToNr-_6gV:-KC0bvRolf6ToNr-_6gW:487584542) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/tests/etl/test_pushlog.py#L44)):*
The `pulse_resultset_consumer` parameter to this function, and `pulse_resultset_consumer()` in conftest.py is now unused :-)

---

*[treeherder/client/thclient/client.py, line 155 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0f-FNrbzbhbmwgZ91:-KC0f-FOGGvJ6i3JahHd:-5152502) ([raw file](https://github.com/mozilla/treeherder/blob/363f8ae1d64b5d015496bbb07c6497534934b4a6/treeherder/client/thclient/client.py#L155)):*
Perhaps worth adding a `logger.warn()` here?

---

*[treeherder/model/tasks.py, line 142 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0dIpkyhyLQbYUTahT:-KC0dIplNhvgzHLe3Ou1:121868809) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/treeherder/model/tasks.py#L142)):*
`get_repository_info()` is now unused - given we're wanting to get rid of datasource maybe not worth keeping it around?

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/1322)*

**Comment by camd** [comment]
Review status: 41 of 48 files reviewed at latest revision, 6 unresolved discussions.

---

*[tests/etl/test_job_loader.py, line 34 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0eX53sTmHsY24uyQU:-KC2EWw2tYSSTdUS2qQk:1679232788) ([raw file](https://github.com/mozilla/treeherder/blob/363f8ae1d64b5d015496bbb07c6497534934b4a6/tests/etl/test_job_loader.py#L34)):*
yeah, that's true.  it's harmless but might as well remove it.  Good spot.  :)

---

*[tests/etl/test_pushlog.py, line 40 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0cMlhIg7Zr1dKIb7y:-KC2ZpI29kDknRFI81wY:-499146009) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/tests/etl/test_pushlog.py#L40)):*
Done.

---

*[tests/etl/test_pushlog.py, line 44 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0bvRolf6ToNr-_6gV:-KC2Zo6Ifa2jra74Sd2L:677588487) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/tests/etl/test_pushlog.py#L44)):*
done

---

*[treeherder/client/thclient/client.py, line 155 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0f-FNrbzbhbmwgZ91:-KC2KD7v0HN8pBrJJJgF:441161852) ([raw file](https://github.com/mozilla/treeherder/blob/363f8ae1d64b5d015496bbb07c6497534934b4a6/treeherder/client/thclient/client.py#L155)):*
good call

---

*[treeherder/model/tasks.py, line 142 \[r2\]](https://reviewable.io:443/reviews/mozilla/treeherder/1322#-KC0dIpkyhyLQbYUTahT:-KC2ZZgw-hxuLEXupbjV:-1330849813) ([raw file](https://github.com/mozilla/treeherder/blob/f99e43e68296600ba8794cccd1fa374025f2def8/treeherder/model/tasks.py#L142)):*
yep, another good call.  I also removed a dup of ``get_all_repository_info()`` which was in there for some reason...

---


*Comments from the [review on Reviewable.io](https://reviewable.io:443/reviews/mozilla/treeherder/1322)*

**Comment by camd** [comment]
@edmorley thanks for the review.  I've made all the changes you suggested.  Thanks for catching all that.  :)

**Pull request merged**